Dockerfile

```
@@ -9,31 +9,43 @@
 #
 # ----------------------------------------------------------------------------------------
 
-FROM python:3.11-slim
+# Builder
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim AS builder
+WORKDIR /app
 
-# Install uv (for fast dependency management)
-RUN pip install --upgrade pip
+# Install git, clone, and cleanup in one layer
+RUN apt-get update && apt-get install -y --no-install-recommends git && \
+    git clone https://github.com/wso2/fhir-mcp-server.git . && \
+    apt-get purge -y --auto-remove git && \
+    rm -rf /var/lib/apt/lists/* /app/.git
+
+# Create venv and install in one layer
+ENV VIRTUAL_ENV=/opt/venv
+ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
+RUN uv venv /opt/venv && \
+    uv pip sync requirements.txt && \
+    uv pip install .
+
+# Runtime
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim AS runtime
+
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    VIRTUAL_ENV=/opt/venv \
+    PATH="/opt/venv/bin:${PATH}" \
+    FHIR_MCP_HOST=0.0.0.0 \
+    FHIR_MCP_PORT=8000 \
+    FHIR_MCP_REQUEST_TIMEOUT=30 \
+    FHIR_SERVER_BASE_URL=http://localhost:8080/fhir
```

This must be provided at runtime

```suggestion
    FHIR_MCP_REQUEST_TIMEOUT=30 
```